classes/Tools.php

```diff
@@ -4307,6 +4309,42 @@ public static function isFileFresh(string $file, int $timeout = self::CACHE_LIFE
 
         return false;
     }
+
+    /**
+     * @return bool
+     */
+    public static function isCountryFromBrowserAvailable(): bool
+    {
+        $languageAvailable = self::getCountryIsoCodeFromHeader();
+
+        if ($languageAvailable === null) {
+            return false;
+        }
+
+        return Configuration::get('PS_DETECT_COUNTRY') && isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])
+            && self::isCountryIsoCodeAvailableInHeader()
+            && Validate::isLanguageIsoCode($languageAvailable);
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isCountryIsoCodeAvailableInHeader(): bool
+    {
+        return preg_match(self::LANGUAGE_EXTRACTOR_REGEXP, $_SERVER['HTTP_ACCEPT_LANGUAGE']);
```

Comment:
Maybe check that the key `HTTP_ACCEPT_LANGUAGE` exists before using it with:
```suggestion
        return preg_match(self::LANGUAGE_EXTRACTOR_REGEXP, $_SERVER['HTTP_ACCEPT_LANGUAGE'] ?? '');
```
or:
```suggestion
        if (!isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
            return false;
        }

        return preg_match(self::LANGUAGE_EXTRACTOR_REGEXP, $_SERVER['HTTP_ACCEPT_LANGUAGE']);
```

Reply:
I prefer the second one ... more explicit